### ISSUE TITLE
Unable to connect to Vault (Error: self signed certificate in certificate chain)

### ISSUE THREAD

**Issue by amej** [owenfarrell/vscode-vault#17]
Visual Studio Code OSS

```
Version: 1.41.1
Commit: 26076a4de974ead31f97692a0d32f90d735645c0
Date: 2020-01-19T05:18:46.103Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Linux x64 4.18.0-80.11.2.el8_0.x86_64
```
Extension version:
`vscode-vault v2.0.0`

The corporate hashicorp vault server is running on the server vault01.util.dc.com.  Port is 8200
The Vault instance does not suport the integrated authentication backends,
I used 'Native' authorization and provided access token.

The vault instance is using self-signed certificates in the  chain.
Hence; in the  'vault.trustedEndpoints'.  I tried out both of these 2 options

1. vault01.util.dc.com
2.  vault01.util.dc.com:8200

Still; I receive the error.  
Unable to connect to Vault (Error: self signed certificate in certificate chain)

**Comment by owenfarrell** [comment]
👋 Hi @amej,

Looks like I forgot to replace the use of the `trustedAuthorities` configuration as a part of the recent UI refresh that I did.

I've coded up a fix for this issue, but haven't gotten around to testing it yet. If you're interested in helping test the changes, you can download the [PR build](https://github.com/owenfarrell/vscode-vault/suites/725673304/artifacts/7229492) and install the VSIX package yourself.

~Owen

**Comment by amej** [comment]
>  Hi @amej,

Hi Owen,

 
> I've coded up a fix for this issue, but haven't gotten around to testing it yet. If you're interested in helping test the changes, you can download the [PR build](https://github.com/owenfarrell/vscode-vault/suites/725673304/artifacts/7229492) and install the VSIX package yourself.

I downloaded the PR build.  Installation fails with the following statement:
_"Unable to install extension 'owenfarrell.vscode-vault' as it is not compatible with VS Code '1.41.1'."_

Following are the technical details -
```

sh-5.0$ unzip artifact.zip 
Archive:  artifact.zip
  inflating: vscode-vault.vsix       
sh-5.0$ ls vscode-vault.vsix 
vscode-vault.vsix
sh-5.0$ file vscode-vault.vsix 
vscode-vault.vsix: Zip archive data, at least v2.0 to extract
sh-5.0$ md5sum vscode-vault.vsix
911b7f1316793163c2e003c5ef2f8028  vscode-vault.vsix
sh-5.0$ 

sh-5.0$ code-oss --install-extension vscode-vault.vsix 
Installing extensions...
(node:464) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(node:464) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Unable to install extension 'owenfarrell.vscode-vault' as it is not compatible with VS Code '1.41.1'.
Failed Installing Extensions: vscode-vault.vsix
```

I am ready to test your proposed fix for this new issue too.

> 
> ~Owen

Ameya

**Comment by owenfarrell** [comment]
Yeah, I upgraded the `vscode` dependencies as part of a [previous commit](https://github.com/owenfarrell/vscode-vault/commit/0ec984181b5d4971624cea0552e138f2e2472723). But I don't think there's anything in the API that I need, so I just reverted that change in this PR.

Could you try the latest [PR build](
https://github.com/owenfarrell/vscode-vault/suites/733930399/artifacts/7381554)? It should be compatible with your version of VSCode.

**Comment by amej** [comment]
> Yeah, I upgraded the `vscode` dependencies as part of a [previous commit](https://github.com/owenfarrell/vscode-vault/commit/0ec984181b5d4971624cea0552e138f2e2472723). But I don't think there's anything in the API that I need, so I just reverted that change in this PR.
> 
> Could you try the latest [PR build](https://github.com/owenfarrell/vscode-vault/suites/733930399/artifacts/7381554)? It should be compatible with your version of VSCode.

@owenfarrell : I tried the latest build [PR build](https://github.com/owenfarrell/vscode-vault/suites/733930399/artifacts/7381554).
Result:  
tldr: **The Hashicorp vault extension connects successfully to the vault01.util.dc.com:8200** 

Details:
I did not encounter the following issue mentioned in the https://github.com/owenfarrell/vscode-vault/issues/17#issuecomment-635817628

> 
> I downloaded the PR build. Installation fails with the following statement:
> "Unable to install extension 'owenfarrell.vscode-vault' as it is not compatible with VS Code '1.41.1'."

The installation is successful.
```
sh-5.0$ code-oss --list-extensions owenfarrell.vscode-vault --show-versions
owenfarrell.vscode-vault@2.0.1
sh-5.0$ 
```

Addition of the servername:port for Native type of authentication   is successful. 

Thank you for the fix.
Regards,
Ameya Sathe

**Comment by owenfarrell** [comment]
@amej - Earlier today, I published an update (2.0.1) to the extension marketplace. You might want to uninstall your custom version and reinstall from the marketplace just to make sure you get future updates.

Thanks for helping validate the fix!
